frappe/model/base_document.py

```diff
@@ -291,12 +295,16 @@ def db_update(self):
 				), d.values() + [d.get("name")])
 		except Exception, e:
 			if e.args[0]==1062:
-				type, value, traceback = sys.exc_info()
-				fieldname = str(e).split("'")[-2]
-				frappe.msgprint(_("{0} must be unique".format(self.meta.get_label(fieldname))))
-				raise frappe.ValidationError, (self.doctype, self.name, e), traceback
+				self.raise_unique_validation(e)
```

Comment:
change function name to `show_unique_validation_message` and use `raise` instead of `raise e` to preserve traceback